.github/workflows/release-pypi.yml

```diff
@@ -37,46 +45,88 @@ jobs:
     strategy:
       matrix:
         # Windows tests fail when built as a binary wheel for some reason
-        os: [ubuntu-latest, macos-latest]  #  windows-latest
+        # 'macos-12' doesn't pass the tests
+        os: [ubuntu-latest, macos-11] # windows-latest
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
 
     # setup Python for cibuildwheel
     - name: Set up Python
-      uses: actions/setup-python@v2
+      uses: actions/setup-python@v4
       with:
         python-version: '3.x'
 
     # for other architectures, see: https://cibuildwheel.readthedocs.io/en/stable/faq/#emulation
     - name: Build wheels for CPython
-      uses: pypa/cibuildwheel@v2.1.1
+      uses: pypa/cibuildwheel@v2.12.3
       env:
-        CIBW_BUILD: 'cp*'  # no PyPI builds
-        CIBW_SKIP: 'cp310-*'  # 3.10 builds are broken
-        CIBW_ARCHS: 'auto64'  # only 64-bit
-        CIBW_TEST_REQUIRES: pytest pytest-xdist
-        CIBW_TEST_COMMAND: pytest -n auto {project}/test/
-
-    - uses: actions/upload-artifact@v2
+        # cibuildwheel automatically reads 'python_requires' from 'setup.py'
+        CIBW_BUILD: 'cp*'  # no PyPy builds
+        CIBW_ARCHS_LINUX: "x86_64" # only Intel 64-bit
+        CIBW_ARCHS_MACOS: "x86_64" # only Intel 64-bit
+        CIBW_TEST_REQUIRES: pytest pytest-xdist filelock setuptools>=60.2
+        # build static libs; those are required for testing
+        CIBW_BEFORE_TEST: python {project}/setup.py build_clib build_ext -i
```

Comment:
I'm a bit unhappy about that and plan to fix that for the next minor release.
`cibuildwheel` is already very clever and changes the working directory to something else than the project root when running the tests. However, it seems that `pytest` discovers the project root and then also prepends it to `sys.path`. As far as I could find out locally, this will still run the tests _from source_ (so, not using the installed HPy package). 

I tried following things to mitigate that:
* Set `--import-mode=append`; fails because with:
  ```text
  ImportError while loading conftest '{project}/test/conftest.py'.
  ModuleNotFoundError: No module named 'test.conftest'
  ```
* Set `--import-mode=importlib`; not possible for us because our test modules have relative imports (as far as I understood from docs)
* Set `--root-dir` to something else; failed

The only thing that worked as expected for me (locally) was to copy `{project}/test` to some temp dir and use that. Maybe that's what we should do here as well but I first need to verify that we can do that.

Do you know what is the proper way to do this with `pytest`?

Reply:
We definitely shouldn't run `setup.py build_XXX` when testing the wheel. It is already built and installed. Perhaps just copy the test folder across to wherever the tests are being run?